src/generic/datavgen.cpp

```diff
@@ -864,7 +864,9 @@ class wxDataViewMainWindow: public wxWindow
     bool EnableDragSource( const wxDataFormat &format );
     bool EnableDropTarget( const wxDataFormat &format );
 
-    void RemoveDropHint();
+    void RemoveDropHint(bool refreshOnly = false);
+    int GetDropItemInfo(const wxCoord x, const wxCoord y,
```

Comment:
This function is unnecessarily complicated: it has a return value and 4 different output parameters. It should just return some struct (`DropItemInfo`?) containing all the fields needed by the caller.

Reply:
```
enum DropHint
{
	DropHint_None = 0,
	DropHint_Inside,
	DropHint_Below,
	DropHint_Above
};
struct DropItemInfo
{
	unsigned int        m_row;
	DropHint            m_hint;

	wxDataViewItem      m_item;
	int                 m_proposedDropIndex;
	int                 m_indentLevel;

	DropItemInfo()
	:	m_row(-1)
	,	m_hint(DropHint_None)
	,	m_item(NULL)
	,	m_proposedDropIndex(-1)
	,	m_indentLevel(-1)
	{
	}
};
```

Is it ok to return such structure by value?
I mean:
`DropItemInfo GetDropItemInfo(const wxCoord x, const wxCoord y)`

Reply:
Yes, sure, why not? This is exactly what I meant as it looks like the simplest solution and should make the code more clear.